### PR TITLE
Fix scons file for arm64 and add mac instructions to readme

### PR DESCRIPTION
**Pull request by marc-weber1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ Features:
 Builds for:
 - [Windows](#windows) ✅
 - [Linux](#linux) ✅
+- [Mac] (#mac) ✅
```
Comment (cgodley) [inline README.md:14]: There should not be a space here
```suggestion
- [Mac](#mac) ✅
```

File: README.md [hunk 2]
```diff
@@ -64,6 +64,16 @@ In thirdparty/opus/build: `make`
 
 In the project root: `scons`
 
+### Mac
+
+Not sure on the status of whether unsigned GDExtensions can be loaded into MacOS, but it builds, so here are the instructions anyways
+
+In thirdparty/opus: `cmake -Bbuild -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64"`
```
Comment (cgodley) [inline README.md:71]: When I tested this on mac m1 Sonoma 14.3, I got the warning `ERRORRuntime cpu capability detection needed for MAY_HAVE_NEON`, and the `make` command failed during `Building C object CMakeFiles/opus.dir/celt/arm/celt_neon_intr.c.o`.

Might want to only build for the local machine (not universal) until that issue is resolved.

```suggestion
In thirdparty/opus: `cmake -Bbuild
```

File: README.md [hunk 3]
```diff
@@ -64,6 +64,16 @@ In thirdparty/opus/build: `make`
...
+
+In thirdparty/opus/build: `make` (?)
```
Comment (cgodley) [inline README.md:73]:
```suggestion
In thirdparty/opus/build: `make`
```

File: SConstruct [hunk 4]
```diff
@@ -20,15 +20,19 @@ sources = []
 if env["platform"] == "windows":
     env.Append(CPPPATH=["thirdparty/opus/include"], LIBS=["thirdparty/opus/build/Release/opus.lib"])
     env.Append(LINKFLAGS = ['/WX:NO'])
-elif env["platform"] == "linux" or env["platform"] == "javascript":
+else:
     env.Append(CPPPATH=["thirdparty/opus/include"], LIBS=["opus"], LIBPATH="thirdparty/opus/build")
     env.Append(CXXFLAGS = ['-fpermissive'])
```
Comment (cgodley) [inline SConstruct:25]: I was able to successfully build and run without `-fpermissive` on mac m1 sonoma 14.3
```suggestion
    if env["platform"] != "macos":
        env.Append(CXXFLAGS = ['-fpermissive'])
```

File: README.md [hunk 5]
```diff
@@ -64,6 +64,16 @@ In thirdparty/opus/build: `make`
 
 In the project root: `scons`
 
+### Mac
+
+Not sure on the status of whether unsigned GDExtensions can be loaded into MacOS, but it builds, so here are the instructions anyways
```
Comment (cgodley) [inline README.md:69]:
```suggestion
Not sure on the status of whether unsigned GDExtensions can be loaded into MacOS, but it builds, so here are the instructions anyways

You may need to set `driver/mix_rate=48000` in `project.godot`
```

**Review by cgodley — CHANGES_REQUESTED** [review]

**Comment by cgodley** [comment]
Sorry for the wall of text, first time using GitHub code review feature

**Comment by marc-weber1** [comment]
tyty

**Comment by marc-weber1** [comment]
w.e I merged it just lmk if there are more changes and ill put them on directly

**Pull request merged**